pallets/pallet-did-lookup/src/lib.rs

```diff
@@ -197,7 +209,22 @@ pub mod pallet {
 				Error::<T>::InsufficientFunds
 			);
 
-			Self::add_association(sender, did_identifier, account)?;
+			match req {
+				AssociateAccountRequest::Substrate(acc, proof) => {
+					ensure!(
+						proof.verify(&get_wrapped_payload(&encoded_payload[..], &acc)[..], &acc),
+						Error::<T>::NotAuthorized
+					);
+					Self::add_association(sender, did_identifier, acc.into())?;
+				}
+				AssociateAccountRequest::Ethereum(acc, proof) => {
+					ensure!(
+						proof.verify(&get_wrapped_payload(&encoded_payload[..], &acc)[..], &acc),
+						Error::<T>::NotAuthorized
+					);
+					Self::add_association(sender, did_identifier, acc.into())?;
+				}
+			};
```

Comment:
can we add this as a verify method to AssociateAccountRequest?

Reply:
Done :)